plugins/inputs/http_response/http_response.go

```diff
@@ -113,18 +119,63 @@ func (h *HTTPResponse) createHttpClient() (*http.Client, error) {
 	return client, nil
 }
 
+func set_result(result_string string, fields *map[string]interface{}, tags *map[string]string) {
+	result_codes := map[string]int{
+		"success":                  0,
+		"response_string_mismatch": 1,
+		"body_read_error":          2,
+		"connection_failed":        3,
+		"timeout":                  4,
+		"dns_error":                5,
+	}
+
+	(*tags)["result"] = result_string
+	(*fields)["result_type"] = result_string
+	(*fields)["result_code"] = result_codes[result_string]
+}
+
+func set_error(err error, fields *map[string]interface{}, tags *map[string]string) error {
+	if timeoutError, ok := err.(net.Error); ok && timeoutError.Timeout() {
```

Comment:
I think this function can be improved by using a [type switch](https://tour.golang.org/methods/16)